### PR TITLE
Added Sr25519Precompile 

### PR DESCRIPTION
**Pull request by akru**
**Pull Request Summary**

This precompile helps to verify Sr25519 signatures in EVM smart contracts.

Proposed interface:

```solidity
interface SR25519 {
    /**
     * @dev Verify signed message using SR25519 crypto.
     * @return A boolean confirming whether the public key is signer for the message. 
     */
    function verify(
        bytes32 public,
        bytes signature,
        bytes message
    ) external view returns (bool);
}
```

**Check list**
- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tests and/or benchmarks are included
- [ ] changed API client type definition or chain metadata

**This pull request makes the following changes:**

**Adds**
- (ex: Add feature A)

**Fixes**
- (ex: Fix validation function)

**Changes**
- (ex: Change document B)

**To-dos**
> *Feel free to remove this section if it's not applicable

- [ ] (ex: add user list)

### CODE REVIEW

File: precompiles/utils/macro/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,144 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
```
Comment (hoonsubin) [inline precompiles/utils/macro/src/lib.rs:4]: You might want to change the headers...

Comment (akru) [reply precompiles/utils/macro/src/lib.rs:4]: Because of use it as is I guess header should also keep for legal reasons.

Comment (Dinonard) [reply precompiles/utils/macro/src/lib.rs:4]: In that case, would it be better to fork their repo and just add our modifications there?

Not that keen on having another `frame` repo to depend on but this header in such form also doesn't look appealing.

Could we just change it to something like "re-used from Moonbeam repository <link>"?

**Comment by Maar-io** [comment]
please change PR to point to branch polkadot-v0.9.16

**Comment by akru** [comment]
@Maar-io changed and resolved conflicts

**Comment by Maar-io** [comment]
@akru test was failing, I added the fix

**Review by Maar-io — APPROVED** [review]
LGTM

**Pull request merged**